src/main/java/aquality/selenium/core/utilities/ElementActionRetrier.java

```diff
@@ -3,52 +3,22 @@
 import aquality.selenium.core.configurations.IRetryConfiguration;
 import com.google.inject.Inject;
 
-import java.util.Optional;
 import java.util.function.Supplier;
 
-public class ElementActionRetrier implements IElementActionRetrier {
-
-    private IRetryConfiguration retryConfiguration;
+public class ElementActionRetrier extends ActionRetrier implements IElementActionRetrier {
 
     @Inject
     public ElementActionRetrier(IRetryConfiguration retryConfiguration) {
-        this.retryConfiguration = retryConfiguration;
+        super(retryConfiguration);
     }
 
     @Override
     public void doWithRetry(Runnable runnable) {
-        Supplier<?> supplier = () -> {
-            runnable.run();
-            return true;
-        };
-        doWithRetry(supplier);
+        doWithRetry(runnable, getHandledExceptions());
     }
 
     @Override
     public <T> T doWithRetry(Supplier<T> function) {
-        int retryAttemptsLeft = retryConfiguration.getNumber();
-        Optional<T> result = Optional.empty();
-        while (retryAttemptsLeft >= 0) {
-            try {
-                result = Optional.of(function.get());
-                break;
-            } catch (Exception exception) {
-                if (isExceptionHandled(exception) && retryAttemptsLeft != 0) {
-                    try {
-                        Thread.sleep(retryConfiguration.getPollingInterval().toMillis());
-                    } catch (InterruptedException e) {
-                        Thread.currentThread().interrupt();
-                    }
-                    retryAttemptsLeft--;
-                } else {
-                    throw exception;
-                }
-            }
-        }
-        return result.orElse(null);
-    }
-
-    protected boolean isExceptionHandled(Exception exception) {
```

Comment:
I believe this isExceptionHandled method was very handy, are you sure that we should remove it?
e.g. I have exception and want to decide handle it or not basing on exception message (like in aquality.appium.mobile for dotnet)
So - can we leave this functionality in ActionRetrier class?

Reply:
Could you please check a fix